UnityProject/Assets/Scripts/Player/PlayerSprites.cs

```diff
@@ -432,34 +433,33 @@ private void UpdateBurningOverlays(float fireStacks, OrientationEnum currentFaci
 
 		public void OnCharacterSettingsChange(CharacterSheet characterSettings)
 		{
-			if (RootBodyPartsLoaded == false)
+			if (RootBodyPartsLoaded) return;
+			RootBodyPartsLoaded = true;
+			var overrideSheet = string.IsNullOrEmpty(RaceOverride) == false;
+			if (characterSettings == null || overrideSheet)
 			{
-				RootBodyPartsLoaded = true;
-				var overrideSheet = string.IsNullOrEmpty(RaceOverride) == false;
-				if (characterSettings == null || overrideSheet)
-				{
-					characterSettings = new CharacterSheet();
-				}
-
-				if (overrideSheet)
-				{
-					characterSettings.Species = RaceOverride;
-				}
+				characterSettings = new CharacterSheet();
+			}
 
-				ThisCharacter = characterSettings;
-				bool notCustomSpecies = string.IsNullOrEmpty(RaceOverride);
-				RaceBodyparts = characterSettings.GetRaceSo(notCustomSpecies);
+			if (overrideSheet)
+			{
+				characterSettings.Species = RaceOverride;
+			}
 
-				if (RaceBodyparts == null)
-				{
-					Logger.LogError($"Failed to find race for {gameObject.ExpensiveName()} with race: {characterSettings.Species}");
-				}
+			ThisCharacter = characterSettings;
+			bool notCustomSpecies = string.IsNullOrEmpty(RaceOverride);
+			RaceBodyparts = gameObject.TryGetComponent<Mob>(out var mob) ?
+				characterSettings.GetRaceSoNoValidation() : characterSettings.GetRaceSo(notCustomSpecies);
 
-				livingHealthMasterBase.InitialiseFromRaceData(RaceBodyparts);
-				livingHealthMasterBase.SetUpCharacter(RaceBodyparts);
-				SetupSprites();
-				livingHealthMasterBase.StartFresh();
+			if (RaceBodyparts == null)
```

Comment:
this shouldn't run if it's null, that shouldn't also auto populate with human, since aliens requires the null